server/manager/src/main/java/org/apache/accumulo/manager/upgrade/PreUpgradeValidation.java

```diff
@@ -117,4 +141,114 @@ private void fail(Exception e) {
     System.exit(1);
   }
 
+  private void validateTableLocks(final ServerContext context) {
+
+    final ZooReaderWriter zrw = context.getZooReaderWriter();
+    final ZooKeeper zk = zrw.getZooKeeper();
+    final String rootPath = context.getZooKeeperRoot();
+    final String tserverLockRoot = rootPath + Constants.ZTSERVERS;
+
+    log.debug("Looking for locks that may be from previous version in path: {}", tserverLockRoot);
+
+    AtomicInteger errorCount = new AtomicInteger(0);
+
+    List<Pair<HostAndPort,ServiceLock.ServiceLockPath>> hostsWithLocks =
+        gatherLocks(zk, tserverLockRoot);
+
+    // try a thrift call to the hosts - hosts running previous versions will fail the call
+    ThreadPoolExecutor lockCheckPool = ThreadPools.getServerThreadPools().createThreadPool(8, 32,
+        10, MINUTES, "update-lock-check", false);
+    try {
+      Future<?> f = lockCheckPool.submit(() -> hostsWithLocks.parallelStream().forEach(p -> {
```

Comment:
This appears that it will submit `1` task to the `lockCheckPool` and that task will then use the common `ForkJoinPool` to check the hosts in parallel. Is that right?

Reply:
The intention is that each parallel task is submitted to the custom pool and then run from there. Maybe I have the syntax wrong?

Reply:
why not just:
```
  hostsWithLocks.forEach( e -> lockCheckPool.submit(new Task(e));
```

I have yet to find something that tells me how the syntax you have here works.

Reply:
reworked in a2839d07c704a8